docs/developers/reproducible-contract-builds.md

```diff
@@ -0,0 +1,203 @@
+codehash_hex = codehash.hex()
+
+print(codehash_hex)
+```
+
+The output is:
+
+```
+8da80dd8bf5b32dfdeaa919901c679d8167162aed7eb7ea41502c7f3d1b27e25
+```
+
+Alternatively, we can locally compute the _codehash_, as well (using the `blake2b` module):
+
+```
+from hashlib import blake2b
+
+import requests
+
+contract = "erd1qqqqqqqqqqqqqpgqahertgz4020wegswus8m7f2ak8a6d0gv396qw3t2zy"
+data = requests.get(f"https://devnet-api.elrond.com/accounts/{contract}").json()
+code_hex = data["code"]
+code = bytes.fromhex(code_hex)
+
+h = blake2b(digest_size=32)
+h.update(code)
+codehash_hex = h.hexdigest()
+
+print(codehash_hex)
+```
+
+If the `WASM` file is directly available, the snippet simplifies to:
+
+```
+from hashlib import blake2b
+
+bytecode_file = open("adder.wasm", "rb")
+code = bytecode_file.read()
+bytecode_file.close()
+
+h = blake2b(digest_size=32)
+h.update(code)
+codehash_hex = h.hexdigest()
+
+print(codehash_hex)
+```
+
+We can also use the utility `b2sum` to locally compute the _codehash_:
+
+```
+b2sum -l 256 adder.wasm
+```
+
+Now, it's easy to verify the bytecode equality of two given builds of a contract using solely the _codehash_ property.
+
+## Supporting reproducible builds
+
+As of September 2022, the recommended approach to support reproducible builds for your smart contract is to perform the **deployment build** (as opposed to regular development builds) in a Docker container based on a [publicly-available, tagged Docker image](https://hub.docker.com/r/elrondnetwork/elrond-sdk-erdpy-rust/tags), that includes tagged, explicit versions of the build tools (_erdpy_, _Rust_, _wasm-opt_ etc.). In addition, the current directory when running the build command should be: `/home/elrond`. 
+
+The above constraints are recommended in order to counteract eventual pieces of non-determinism related to `cargo`'s (essential component of the Rust toolchain) sensibility on the environment.
+
+:::important
+If the code source of your smart contract is hosted on GitHub, then it's a good practice to define a GitHub Workflow similar to [this one](https://github.com/ElrondNetwork/reproducible-contract-build-example/blob/main/.github/workflows/release.yml), which performs the deployment (production-ready) build within the _release_ procedure.
+:::
+
+### Choose an image tag
+
+The first step for supporting reproducible builds is to decide on a specific Docker image tag to use. Check the **frozen** tags listed at [elrondnetwork/elrond-sdk-erdpy-rust](https://hub.docker.com/r/elrondnetwork/elrond-sdk-erdpy-rust/tags), and inspect their labels - especially the label `rust`:
+
+```
+LABEL rust=nightly-2022-03-01
+```
+
+The chosen, _frozen_ image tag should accompany the versioned source code (e.g. via _release notes_), in order to inform others on how to reproduce a specific build (of a specific source code version). In this context, _frozen_ image tag refers to a Docker image tag that will never get any updates after its initial publishing.
+
+:::tip
```

Comment:
Maybe also shortly explain the difference between them?

Reason:
Right now we are saying: There are multiple versions, if you don't know what you are doing, pick the largest one.
- But what if I want to know what am I doing?

Reply:
Added extra info.